packages/amplify-provider-awscloudformation/src/aws-utils/aws-s3.ts

```diff
@@ -13,6 +13,8 @@ const minChunkSize = 5 * 1024 * 1024; // 5 MB https://docs.aws.amazon.com/AWSJav
 const { fileLogger } = require('../utils/aws-logger');
 const logger = fileLogger('aws-s3');
 
+type OptionalExceptFor<T, TRequired extends keyof T> = Partial<T> & Pick<T, TRequired>
```

Comment:
gettin fancy with the ctrl+C ctrl+V I see: https://stackoverflow.com/questions/52703321/make-some-properties-optional-in-a-typescript-type ;)